pagefaultgames/pokerogue
# [enhancement] Added overrides for Post Battle rewards

## What are the changes?
Added the ability to override how many and what items appear during the post battle reward screen.

## Why am I doing these changes?
There are a couple thing I want to do at some point where having the ability to specify rewards will be very helpful and I can also imagine others might find this functionality useful. So, I figure it is probably preferable to get this change reviewed/committed first separately from the other stuff I will be doing.

## What did change?
I added a REWARD_AMOUNT_OVERRIDE value that can be used to override the amount of rewards that appears in the selection, and REWARD_OVERRIDES that can override some or all of the rewards that appear.

I also, instead of having the override code in the place that uses it, put it in the override.ts file. I think once this is merged, something I will look to do is move any overriding code to overrides so that it can all be in one place rather than scattered throughout the codebase.

### Screenshots/Videos
Example shot showing it is the first wave, but I have overridden it to show 5 rewards and the first two rewards is the Acrobatics TM and Lum Berry.

## How to test the changes?
Go into overrides.ts and change the REWARD_AMOUNT_OVERRIDE and/or REWARD_OVERRIDES. Then win a battle and see the rewards use the values you specify.

## Checklist
